src/pages/workspace/withPolicyAndFullscreenLoading.tsx

```diff
@@ -0,0 +1,45 @@
+import isEmpty from 'lodash/isEmpty';
+import React, {ComponentType, ForwardedRef, forwardRef, RefAttributes} from 'react';
+import {OnyxEntry, withOnyx} from 'react-native-onyx';
+import FullscreenLoadingIndicator from '@components/FullscreenLoadingIndicator';
+import compose from '@libs/compose';
+import ONYXKEYS from '@src/ONYXKEYS';
+import withPolicy, {WithPolicyOnyxProps, WithPolicyProps} from './withPolicy';
+
+type WithPolicyAndFullscreenLoadingOnyxProps = {
+    /** Indicated whether the report data is loading */
+    isLoadingReportData: OnyxEntry<boolean>;
+};
+
+type WithPolicyAndFullscreenLoadingProps = WithPolicyProps & WithPolicyAndFullscreenLoadingOnyxProps;
+
+export default function withPolicyAndFullscreenLoading<TProps extends WithPolicyAndFullscreenLoadingProps, TRef>(
+    WrappedComponent: ComponentType<TProps & RefAttributes<TRef>>,
+): React.ComponentType<Omit<Omit<TProps & React.RefAttributes<TRef>, 'isLoadingReportData'>, keyof WithPolicyOnyxProps>> {
+    function WithPolicyAndFullscreenLoading(props: TProps, ref: ForwardedRef<TRef>) {
```

Comment:
NAB: I would probably use destructuring on props, but considering that you later spread them in returned component it's also a good solution to leave it like that